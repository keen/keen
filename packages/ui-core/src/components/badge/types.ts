import { colors } from '@keen.io/colors';

export type Variant = keyof typeof colors;

export type TruncateMethod = 'programmatical' | 'css';
