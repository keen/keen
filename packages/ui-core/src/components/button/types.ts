export type ButtonSize = 'default' | 'small' | 'large';

export type ButtonType = 'solid' | 'outline';
