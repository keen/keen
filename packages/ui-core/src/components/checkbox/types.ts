export type Variant = 'primary' | 'secondary' | 'highlight';
