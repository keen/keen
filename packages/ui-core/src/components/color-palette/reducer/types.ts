import {
  SET_ACTIVE_COLOR,
  SET_ADD_COLOR_PICKER_OPEN,
  SET_COLOR_PICKER_POSITION,
  SET_COLORS,
  SET_IS_DRAGGED,
} from './constants';

interface SetIsDragged {
  type: typeof SET_IS_DRAGGED;
  payload: {
    isDragged: boolean;
  };
}

interface SetColors {
  type: typeof SET_COLORS;
  payload: {
    colors: string[];
  };
}

interface SetActiveColor {
  type: typeof SET_ACTIVE_COLOR;
  payload: {
    color: string;
  };
}

interface SetAddColorPickerOpen {
  type: typeof SET_ADD_COLOR_PICKER_OPEN;
  payload: {
    addColorPickerOpen: boolean;
  };
}

interface SetColorPickerPosition {
  type: typeof SET_COLOR_PICKER_POSITION;
  payload: {
    colorPickerPosition: { x: number; y: number };
  };
}

export type ColorPaletteActions =
  | SetIsDragged
  | SetColors
  | SetActiveColor
  | SetAddColorPickerOpen
  | SetColorPickerPosition;
