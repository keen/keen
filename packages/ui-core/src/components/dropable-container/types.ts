export type DropableContainerVariant = 'primary' | 'secondary' | 'transparent';
