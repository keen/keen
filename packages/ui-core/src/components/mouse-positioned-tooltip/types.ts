export type TooltipPinPlacements = 'top-left' | 'top-right' | 'bottom-left';
