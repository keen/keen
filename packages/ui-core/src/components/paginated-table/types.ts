import { Formatter, PatternFormatterDataType } from '@keen.io/charts-utils';

export type CellValue = string | number | boolean | Date | string[] | number[];
export type DataSelector = (number | string)[];
export type CellTextAlignment = 'left' | 'right';

export type TooltipState = {
  visible: boolean;
  selectors: { selector: DataSelector; color: string }[];
  x: number;
  y: number;
};

export type FormattedValue = {
  value: string | number | boolean | Date;
  formatterType?: PatternFormatterDataType;
};

export type ValueFormatter = Record<string, Formatter>;

export type ColumnSelection = {
  name: string;
  formatter: string | null;
  dataType: 'string' | 'number' | 'datetime' | 'boolean';
};

export type ColumnsSelectionEvent = {
  eventName: '@table/columns-selected';
  meta: { selection: ColumnSelection[] };
};

export type TableEvents =
  | ColumnsSelectionEvent
  | { eventName: '@table/deselect-columns' };
