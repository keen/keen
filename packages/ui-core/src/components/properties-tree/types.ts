import { FixedSizeNodeData } from 'react-vtree';

export type DataNode = Readonly<{
  children: Record<string, any>;
  id: string;
  name: string;
}>;

export type StackElement = Readonly<{
  deepnessLevel: number;
  node: DataNode;
}>;

export type TreeData = FixedSizeNodeData &
  Readonly<{
    isLeaf: boolean;
    name: string;
    deepnessLevel: number;
    schemaMeta: any;
  }>;
