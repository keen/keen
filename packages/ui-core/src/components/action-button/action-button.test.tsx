import React, { ComponentProps } from 'react';
import {
  render as rtlRender,
  fireEvent,
  getNodeText,
} from '@testing-library/react';
import 'jest-styled-components';

import ActionButton from './action-button.component';
import { ButtonAction } from './types';

const render = (
  overProps: Partial<ComponentProps<typeof ActionButton>> = {}
) => {
  const props = {
    action: 'create' as ButtonAction,
    ...overProps,
  };
  const wrapper = rtlRender(<ActionButton {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('@keen.io/ui-core - <ActionButton />', () => {
  test('should render action button component', () => {
    const {
      wrapper: { getByTestId },
    } = render();
    const element = getByTestId('action-button');

    expect(element).toBeInTheDocument();
  });

  test('should call onClick callback', () => {
    const mockFn = jest.fn();
    const {
      wrapper: { getByTestId },
    } = render({ onClick: mockFn });
    const element = getByTestId('action-button');
    fireEvent.click(element);

    expect(mockFn).toHaveBeenCalled();
  });

  test('should not call onClick callback once button is disabled', () => {
    const mockFn = jest.fn();
    const {
      wrapper: { getByTestId },
    } = render({ onClick: mockFn, isDisabled: true });
    const element = getByTestId('action-button');
    fireEvent.click(element);

    expect(mockFn).not.toHaveBeenCalled();
  });

  test('should render create button icon', () => {
    const {
      wrapper: { getByTestId },
    } = render();
    const element = getByTestId('action-icon');
    const nodeText = getNodeText(element);

    expect(nodeText).toEqual('+');
  });

  test('should render remove button icon', () => {
    const {
      wrapper: { getByTestId },
    } = render({ action: 'remove' });
    const element = getByTestId('action-icon');
    const nodeText = getNodeText(element);

    expect(nodeText).toEqual('Ã—');
  });

  test('should render custom border radius', () => {
    const {
      wrapper: { getByTestId },
    } = render({ borderRadius: '0' });
    const element = getByTestId('action-button');

    expect(element).toMatchSnapshot();
  });

  test('should not render background', () => {
    const {
      wrapper: { getByTestId },
    } = render({ background: 'red' });
    const element = getByTestId('action-icon');

    expect(element).toMatchSnapshot();
  });

  test('should add custom class', () => {
    const { container } = rtlRender(
      <ActionButton className="custom-button" action="create" />
    );
    const element = container.querySelector('.custom-button');

    expect(element).toBeInTheDocument();
  });
});
