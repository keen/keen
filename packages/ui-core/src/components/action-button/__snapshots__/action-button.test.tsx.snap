// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@keen.io/ui-core - <ActionButton /> should not render background 1`] = `
.c0 {
  font-family: 'Lato Regular',sans-serif;
  font-size: 25px;
  line-height: 30px;
  color: #487650;
}

<span
  action="create"
  class="c0"
  data-testid="action-icon"
>
  +
</span>
`;

exports[`@keen.io/ui-core - <ActionButton /> should render custom border radius 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 0;
  background-color: rgba(119,163,187,0.15);
  padding: 5px 11px;
  width: 37px;
  height: 37px;
  outline: none;
  border: none;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 0.15s ease-in-out;
  transition: background-color 0.15s ease-in-out;
}

.c0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(119,163,187,0.25);
}

.c0:focus-visible {
  outline: 1px solid #85B4C3;
}

.c1 {
  font-family: 'Lato Regular',sans-serif;
  font-size: 25px;
  line-height: 30px;
  color: #487650;
}

<button
  class="c0"
  data-testid="action-button"
  type="button"
>
  <span
    action="create"
    class="c1"
    data-testid="action-icon"
  >
    +
  </span>
</button>
`;
